docs/source/notes/numpy_compatibility.rst

```diff
@@ -0,0 +1,388 @@
+.. _numpy_compatibility:
+
+.. testsetup:: *
+
+   import numpy as np
+   import torch
+
+NumPy Compatibility
+===================
+
+.. note::
+
+   **TL;DR:**
+
+   - Be explicit about obtaining and using NumPy arrays
+   - `This tutorial`_ shows a more concrete example of interoperability between
+     NumPy and PyTorch
+
+At the onset it should be stressed that PyTorch tensors and NumPy arrays are
+fundamentally different types. However, in the interests of being accessible to
+the wider scientific python ecosystem, there are implicit and explicit
+conversion methods. The major considerations to keep in mind are:
+
+- **Where** is the structure allocated?
+    * This may be on a compute device (e.g. a GPU) or the CPU
+- **What** is the underlying data?
+    * In particular, several functions will either return a copy or a view
+- **How** are modifications to the data handled?
+    * The data returned may be read-only, or may be writable
+
+From the end-user perspective, it is most desirable to have zero copy methods,
+however, for exploratory analyses and their visualizations in particular, it is
+often more important to have guaranteed conversions, even if copies are created.
+
+Basic interoperability
```

Comment:
How and when tensors share memory is interesting, too. Should this cover converting NumPy arrays to torch tensors and converting torch tensors to NumPy arrays?

Reply:
Added a section on views along with more examples of the conversions.